planetiler-core/src/main/java/com/onthegomap/planetiler/util/Wikidata.java

```diff
@@ -300,10 +300,15 @@ private LongObjectMap<Map<String, String>> queryWikidata(List<Long> qidsToFetch)
       .POST(HttpRequest.BodyPublishers.ofString(query, StandardCharsets.UTF_8))
       .build();
 
-    InputStream response = null;
-    for (int i = 0; i <= config.httpRetries() && response == null; i++) {
+    LongObjectMap<Map<String, String>> result = null;
+    for (int i = 0; i <= config.httpRetries() && result == null; i++) {
       try {
-        response = client.send(request);
+        var response = client.send(request);
+        if (response != null) {
+          try (var bis = new BufferedInputStream(response)) {
+            result = parseResults(bis);
+          }
+        }
       } catch (IOException e) {
         boolean lastTry = i == config.httpRetries();
         if (!lastTry) {
```

Comment:
While we're at it, can we add a sleep (maybe 5-10s?) here to back off on the server a bit when there's an error. The GOAWAY response makes it sound like the server wants us to slow down.

Reply:
Adjusted.